src/libcollections/str.rs

```diff
@@ -164,17 +164,27 @@ impl ToOwned for str {
     }
 }
 
-/// Any string that can be represented as a slice.
+/// Methods for string slices.
 #[lang = "str"]
 #[cfg(not(test))]
 impl str {
-    /// Returns the length of `self` in bytes.
+    /// Returns the length of `self`.
+    ///
+    /// This length is in bytes, not [`char`]s, or graphemes. In other words,
+    /// it may not be what a human considers the length of the string.
```

Comment:
I said 'may' for two reasons:

1. ASCII strings will be, which are common
2. I don't want to assume what said person knows, so it's a hedge.